migrations/back-37/Hasher.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"strconv"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+
+	"github.com/tidepool-org/platform/data/deduplicator/deduplicator"
+	"github.com/tidepool-org/platform/data/types"
+)
+
+// used for BACK-37 to set hash for jellyfish datum for migrating to platform API
+func CreateHash(bsonData bson.M) string {
+	// represents data from base id fields
+	var baseData = func(bsonData bson.M) []string {
+		return []string{
+			bsonData["_userId"].(string),
```

Comment:
[nit] I wouldn't be surprised if there is old data in the production database that is missing one or more of these fields. If there is, then I believe the forced typecast will panic and crash the migration process. Do we know with 100% certainty that these four fields are on every record and are of the expected type?

Reply:
Also, even if the fields are there, you may want to do the same checks as in `Base.IdentifyFields()` to ensure there aren't any missing strings or zero times.

Reply:
Same down below in the `switch` statement.